spec/unit/helpers_shared_spec.rb

```diff
@@ -74,4 +74,60 @@
       expect(read_config).to match_array([azure_heart_beat_file,azure_status_folder,azure_plugin_log_location,azure_config_folder,azure_status_file])
     end
   end
+end
+
+describe ChefAzure::DeleteNode do
+  include ChefAzure::Shared
+  before do
+
+  end
+
+  let(:dummy_class) { Class.new { extend ChefAzure::DeleteNode } }
+
+  context "delete_node" do
+    it "deletes node and client from chef server" do
+      allow(dummy_class).to receive(:bootstrap_directory).and_return('/etc/chef')
+      allow(Chef::Config).to receive(:from_file).and_return("")
+      Chef::Config[:node_name] = 'foo'
+      allow(Chef::Node).to receive(:load).with('foo').and_return(double(:destroy=>true))
+      allow(Chef::ApiClient).to receive(:load).with('foo').and_return(double(:destroy=>true))
+      expect(dummy_class.delete_node).to eq(0)
+    end
+
+    it 'return exit code 1 if node name not set' do
+      allow(dummy_class).to receive(:bootstrap_directory).and_return('/etc/chef')
+      allow(Chef::Config).to receive(:from_file).and_return("")
+      expect(dummy_class.delete_node).to eq(1)
+    end
+
+    it 'set node_name as fqdn/hostname if node net is not set in configuration file' do
```

Comment:
In above spec title ` if node net is` has typo Please rename `net` to `name`

I think you can give better name to above spec like- use `fqdn` as node_name while deleting node from chef-server ...